### PR TITLE
Trim in-repo copy of Martor

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Also stop checking if a request is_ajax since this is now deprecated and was supposedly never a reliable indicator whether a request was ajax or not.

### CODE REVIEW

File: martor/settings.py [hunk 1]
```diff
@@ -4,7 +4,6 @@
 # Input: string boolean, `true/false`
 MARTOR_ENABLE_CONFIGS = getattr(
     settings, 'MARTOR_ENABLE_CONFIGS', {
-        'emoji': 'true',        # to enable/disable emoji icons.
         'imgur': 'true',        # to enable/disable imgur/custom uploader.
```
Comment (kiritofeng) [inline martor/settings.py:7]: Should we also need to remove `imgur` and `mention`?

Comment (Ninjaclasher) [reply martor/settings.py:7]: They're actually being used - imgur is actually generic custom image uploader and mention is for allowing search when you have `[user:<cursor>]`.

Comment (quantum5) [reply martor/settings.py:7]: If they are being used, is it safe to delete the extensions?

File: martor/static/martor/js/martor.js [hunk 2]
```diff
@@ -145,9 +127,9 @@
             }
             // Set autocomplete for ace editor
             if (editorConfig.mention === 'true') {
-                editor.completers = [emojiWordCompleter, mentionWordCompleter]
+                editor.completers = [mentionWordCompleter]
             }else {
```
Comment (quantum5) [inline martor/static/martor/js/martor.js:131]: Let's autoformat this js at some point... Disgusting...

**Review by quantum5 — COMMENTED** [review]
LGTM, one final question.

File: martor/settings.py (hunk 1 above)
Comment (Ninjaclasher) [reply martor/settings.py:7]: Unless I am mistaken, I believe extensions are only used if we use their markdownify function (https://github.com/DMOJ/online-judge/blob/master/martor/utils.py#L29), which I don't forsee any plans to. Other than that, they are unused.

**Review by quantum5 — APPROVED** [review]

File: martor/static/martor/js/martor.js (hunk 2 above)
Comment (int-y1) [reply martor/static/martor/js/martor.js:131]: #2388

**Pull request merged**